examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,293 @@
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+// The macro compares all the leaves of two TTree objects.
+// The first and strongest comparison is that the entries in the two leaves
+// to compare  are idential.
+// To check this one loops over the entries in the trees, calculate the
+// difference between the two values and fills the result in a histogram.
+// If all entries are absolutely identical the result in the histogram is a
+// delta function at 0.
+// If the first check for identical entries fails it is checked if there are
+// only some changes in the order of the entries. In this case the two
+// produced histgrams (one for each tree) are identical.
+// The last check which again is only done if both previous test fails uses a
+// Kolmogorov test to check if the produced hostograms are comparable on a
+// statistical base.
+// If at least for one leaf all theree comparisons fail the complete test
+// fails.
+
+#include <RtypesCore.h>
+#include <TBranch.h>
+#include <TFile.h>
+#include <TH1.h>
+#include <TMath.h>
+#include <TObjArray.h>
+#include <TString.h>
+#include <TTree.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+std::vector<TString> GetLeafNames(TTree&);
+
+bool operator==(TH1 const&, TH1 const&);
+
+int TreeCompareAuto(TString fileName1 = "", TString fileName2 = "")
+{
+    if (fileName1.IsNull() || fileName2.IsNull()) {
+        cout << "Filenames are not defined" << endl;
+        exit(42);
+    }
+
+    // Get the output tree from the original file
+    std::unique_ptr<TFile> file1{TFile::Open(fileName1, "READ")};
+    if (!file1) {
+        std::cout << "Could not open file " << fileName1 << std::endl;
+        return 42;
+    }
+    std::unique_ptr<TTree> tree1{file1->Get<TTree>("cbmsim")};
+    if (!tree1) {
+        std::cout << "File " << fileName1 << " does not have the cbmsim tree." << std::endl;
+        return 42;
+    }
+
+    // Get the output tree from the file which should be compared
+    std::unique_ptr<TFile> file2{TFile::Open(fileName2, "READ")};
+    if (!file2) {
+        std::cout << "Could not open file " << fileName2 << std::endl;
+        return 42;
+    }
+    std::unique_ptr<TTree> tree2{file2->Get<TTree>("cbmsim")};
+    if (!tree2) {
+        std::cout << "File " << fileName2 << " does not have the cbmsim tree." << std::endl;
+        return 42;
+    }
+
+    // Add the output tree from the file to compare as friend to the tree
+    // of the original file. This allows to access a data member of the
+    // original file by e.g. StsHit.fX and the data element of the second tree
+    // by tree2.StsHit.fX
+    tree1->AddFriend(tree2.get(), "tree2");
+
+    // Get the leaf names for all leaves which should be compared
+    // from the first input tree
+    auto leaves = GetLeafNames(*tree1.get());
```

Comment:
```suggestion
    auto leaves = GetLeafNames(*tree1);
```

See [`unique_ptr::operator*`](https://en.cppreference.com/w/cpp/memory/unique_ptr/operator*)

Reply:
I was sure I tested it and it didn't work, but you are right it works as expected. Fixed.